toolchain/check/import_cpp.cpp

```diff
@@ -70,13 +68,81 @@ auto ImportCppFiles(
         IntAsSelect, IntAsSelect, IntAsSelect, std::string);
     context.emitter().Emit(loc, CppInteropParseError, num_errors, num_warnings,
                            num_imports, diagnostics_str);
-  } else if (num_warnings > 0) {
+    return std::nullopt;
+  }
+  if (num_warnings > 0) {
     CARBON_DIAGNOSTIC(CppInteropParseWarning, Warning,
                       "{0} warning{0:s} in `Cpp` {1} import{1:s}:\n{2}",
                       IntAsSelect, IntAsSelect, std::string);
     context.emitter().Emit(loc, CppInteropParseWarning, num_warnings,
                            num_imports, diagnostics_str);
   }
+  return ast;
+}
+
+static auto AddNamespace(Context& context, IdentifierId cpp_package_id,
+                         SemIR::InstId first_import_decl_id) -> void {
+  // TODO: Extract and deudplicate common logic with import.cpp.
+  auto name_id = SemIR::NameId::ForIdentifier(cpp_package_id);
+  SemIR::TypeId namespace_type_id =
+      context.GetSingletonType(SemIR::NamespaceType::SingletonInstId);
+
+  auto* parent_scope = &context.name_scopes().Get(SemIR::NameScopeId::Package);
+  auto [inserted, entry_id] = parent_scope->LookupOrAdd(
+      name_id,
+      // This InstId is temporary and would be overridden if used.
+      first_import_decl_id, SemIR::AccessKind::Public);
+  CARBON_CHECK(inserted);
+
+  CARBON_CHECK(first_import_decl_id.has_value());
+  auto import_loc_id = context.insts().GetLocId(first_import_decl_id);
+
+  auto namespace_inst = SemIR::Namespace{
+      namespace_type_id, SemIR::NameScopeId::None, first_import_decl_id};
```

Comment:
I believe we'd like the fields to be named in here: https://github.com/carbon-language/carbon-lang/blob/133717cd7e5b821fa832cad2ad85be95ba47ce37/docs/project/cpp_style_guide.md?plain=1#L132
(I see import.cpp didn't name them here, but does in other cases, probably an oversight/inconsistency there.)

Reply:
Done.